### PR TITLE
[DOCS] Document EQL search REST API

### PR DESCRIPTION
**Pull request by jrodewig**
Add API reference documentation for the EQL search API added with #49768.

For now, this PR only documents parameters expected for use with milestone 1 of #49581. Parameters used for joins are marked as `Reserved for future use`.

Remaining parameters and response will be documented alongside future development.

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by matriv** [comment]
@elastic/es-ql

**Review by astefan — APPROVED** [review]
LGTM. Left two minor comments.

File: docs/reference/eql/eql-search-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+[[eql-search-api-path-params]]
+==== {api-path-parms-title}
+
+`<index>`::
+(Required, string)
+Comma-separated list of index names or <<indices-aliases,index aliases>> used to
+limit the request. Accepts wildcard expressions.
++
+To search all indices, use `_all`.
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/eql/eql-search-api.asciidoc]: Addressed this with d09b28600e254be8d69569be89ca3c62301d4ca7. Thanks!

File: docs/reference/eql/eql-search-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+(Optional, <<query-dsl,query DSL object>>)
+Query, written in query DSL, used to filter the documents on which the EQL query
+runs.
+
+`search_after`::
+(Optional, string)
+Reserved for future use.
+
+`size`::
+(Optional, integer)
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/eql/eql-search-api.asciidoc]: Added float with d09b28600e254be8d69569be89ca3c62301d4ca7. Thanks!

**Comment by jrodewig** [comment]
Thank you @astefan.

**Comment by jrodewig** [comment]
### Backport commits
master 32317a6
7.x c1b0548

**Pull request merged**